cli/src/command_error.rs

```diff
@@ -554,16 +554,35 @@ jj currently does not support partial clones. To use jj with this repository, tr
                     err,
                     "Run `jj git remote rename` to give a different name.",
                 ),
-                GitPushError::RefInUnexpectedLocation(refs) => user_error_with_hint(
-                    format!(
-                        "Refusing to push a bookmark that unexpectedly moved on the remote. \
-                         Affected refs: {}",
-                        refs.join(", ")
-                    ),
-                    "Try fetching from the remote, then make the bookmark point to where you want \
-                     it to be, and push again.",
-                ),
-                GitPushError::RefUpdateRejected(_) => user_error(err),
+                GitPushError::FailedRefs {
+                    unexpected_locations,
+                    rejected_updates,
+                } => {
+                    let reference_errors = unexpected_locations
+                        .iter()
+                        .map(|x| format!("{x}: unexpectedly moved on the remote"))
+                        .chain(rejected_updates.iter().map(|x| {
+                            format!("{x}: remote rejected (do you have permission to push?)")
+                        }))
+                        .join("\n");
```

Comment:
Maybe we can add a function (similar to `git_util::print_failed_git_export()`) which formats failed refs and reasons to `&mut dyn Formatter`. Then, the formatted output can be embedded as a hint by `.add_formatted_hint_with()`. It isn't actually a hint, but we don't have support for multi-line/labeled error messages.

If we add support for partial push, failed refs might be printed as a warning.